### PR TITLE
New version of .eln and reading sampleDB and kadi4mat files

### PR DESCRIPTION
**Pull request by SteffenBrinckmann**

### CODE REVIEW

**Comment by SteffenBrinckmann** [comment]
@jmurugan-fzj  this should work for all .eln files which currently fulfil the .eln standard. kadi4mat and sampleDB

File: pasta_eln/inputOutput.py [hunk 1]
```diff
@@ -122,8 +195,8 @@ def processPart(part:dict[str,str]) -> int:
       if not isinstance(part, dict): #leave these tests in since other .elns might do funky stuff
         print("**ERROR in part",part)
         return False
-      logging.info('Process: '+part['@id'])
-      print('Process: '+part['@id'])
+      logging.info('\nProcess: '+part['@id'])
+      print('\nProcess: '+part['@id'])
```
Comment (jmurugan-fzj) [inline pasta_eln/inputOutput.py:199]: Already logged, why do we need this print?

File: pasta_eln/inputOutput.py [hunk 2]
```diff
@@ -144,8 +217,8 @@ def processPart(part:dict[str,str]) -> int:
         elif elnName == 'eLabFTW':
           supplementalInfo = Path(dirName)/elnID/'export-elabftw.json'
         else:
-          print('**ERROR could not identify elnName', elnName)
-          return -1
+          print('**Info: no additional information in', elnName)
```
Comment (jmurugan-fzj) [inline pasta_eln/inputOutput.py:220]: Please change to log instead of print

**Review by jmurugan-fzj — APPROVED** [review]

File: pasta_eln/inputOutput.py [hunk 3]
```diff
@@ -28,22 +35,84 @@
...
-  'tags':'-tags',
-  'lastchange':'-date',
-  'userid':'-user',
-  'metadata':'metaUser',
-  'id':'_id',
-  'category': '-type',
-  'dateCreated': '-date'
+specialTerms:dict[str,dict[str,str]] = {
+    'elabFTW': {
+      # 'elabid':'_id',
```
[the remark this answers is not in the document]
Comment (SteffenBrinckmann) [reply pasta_eln/inputOutput.py:44]: elabftw is not finished with the final implementation; we wait for them to finish; then we adopt this part

**Pull request merged**